lib/blackbeard/context.rb

```diff
@@ -53,15 +53,26 @@ def unique_identifier
     end
 
     def user_id
+      unless bbd.nil?
+        VisitorUserTracker.set_visitor_id_for_user(user_id: @user.id, visitor_id: bbd)
+      end
+
+      vid = VisitorUserTracker.get_visitor_id_for_user(@user.id)
+      return vid if vid
       @user.id
     end
 
     def visitor_id
-      @visitor_id ||= (controller.request.cookies['bbd'] || generate_visitor_id).to_i
+      @visitor_id ||= (bbd || generate_visitor_id).to_i
     end
 
 private
 
+    def bbd
```

Comment:
The cookie is called bbd to minimize the number of bytes in each request. It's not a meaningful enough name for a method.